perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -256,11 +278,13 @@ def Create(self):
       if self.subnet_resource:
         self.subnet_resource.Create()
       self.default_firewall_rule.Create()
+      self.default_firewall_rule2.Create()
```

Comment:
always gonna create second firewall rule?

Reply:
Possibly?

As it's written now, `default_firewall_rule` allows the entire ClassA network `10.0.0.0/8` in GCE. I first thought this was for the default GCE VPC range (auto mode), but that only lives in the top half `10.128.0.0/9` [src](https://cloud.google.com/vpc/docs/vpc#ip-ranges). Then over in `aws_network.py` PKB carves up a `10.0.0.0/16` even though the AWS default VPC is in `172.31.0.0/16` [src](https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html).   If the intent is to open up all RFC1918 private addresses then we should probably have `NETWORK_RANGE[A|B|C]= [10.0.0.0/8|172.16.0.0/12|192.168.0.0/16]`. If not then maybe `NETWORK_RANGE` can be trimmed to `10.0.0.0/9` or `10.128.0.0/9` accordingly.


This 3rd [commit](https://github.com/SMU-ATT-Center-for-Virtualization/PerfKitBenchmarker/commit/7d05fe9b66c06c3583f5bfd531373b11714b126a) puts in a check that only opens ranges in use by the benchmark, so adding a single default AWS network in `172.31.1.0/24`   works through the benchmark config without opening up everything before hand. I've tested with pairs in the default ranges (auto mode), custom ranges (--cidr), and with one endpoint in each. Auto mode falls back on the existing behaviour (open all) listed above to avoid any breakage.